src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -2483,25 +2483,42 @@ void EmitSingleCharFixedRepeater(RegexNode node, bool emitLengthCheck = true)
                     return;
                 }
 
-                // if ((uint)(textSpanPos + iterations - 1) >= (uint)textSpan.Length) goto doneLabel;
-                if (emitLengthCheck)
-                {
-                    EmitSpanLengthCheck(iterations);
-                }
-
                 if (iterations <= MaxUnrollSize)
                 {
-                    // if (textSpan[textSpanPos] != c1 ||
+                    // if ((uint)(textSpanPos + iterations - 1) >= (uint)textSpan.Length ||
+                    //     textSpan[textSpanPos] != c1 ||
                     //     textSpan[textSpanPos + 1] != c2 ||
                     //     ...)
-                    //       goto doneLabel;
-                    for (int i = 0; i < iterations; i++)
+                    // {
+                    //     goto doneLabel;
+                    // }
+                    writer.Write($"if (");
+                    if (emitLengthCheck)
                     {
-                        EmitSingleChar(node, emitLengthCheck: false);
+                        writer.WriteLine($"{SpanLengthCheck(iterations)} ||");
+                        writer.Write("    ");
```

Comment:
As an aside comment, I've seen in some places in the emitter we just do:
```c#
writer.Indent += 4;
// .. your indented writes
writer.Indent -= 4;
```

And we also have few places where we just write the spaces manually. If you agree, I'm happy to put up a PR that just normalizes to always use the Indent property to be consistent.

Reply:
I'm fine with whichever is more readable and maintainable.